frontend/app/src/components/form/fields/relationship.field.tsx

```diff
@@ -12,6 +12,25 @@ import {
 } from "@/components/form/type";
 import { LabelFormField } from "@/components/form/fields/common";
 import { updateRelationshipFieldValue } from "@/components/form/utils/updateFormFieldValue";
+import useQuery from "@/hooks/useQuery";
+import { getRelationshipParent } from "@/graphql/queries/objects/getRelationshipParent";
+import { gql } from "@apollo/client";
+
+const getGenericParentRelationship = (kind?: string) => {
+  if (!kind) return;
```

Comment:
in both these function, checking kind presence is better to do outside of it (before calling the function)